### PR TITLE
Fix ccdb access; implement track matching

### PR DESCRIPTION
**Pull request by peressounko**

### CODE REVIEW

**Comment by peressounko** [comment]
dear @ddobrigk could you please have a look at this PR?

**Comment by peressounko** [comment]
Dear @mikesas could you please approve this PR?

**Comment by peressounko** [comment]
Dear @ktf, could you please approve this PR?

**Comment by peressounko** [comment]
Dear @mikesas could you please approve this PR?

**Review by mikesas — APPROVED** [review]

**Review by ddobrigk — APPROVED** [review]
Looks okay to me, thanks a lot!

File: Common/TableProducer/caloClusterProducer.cxx [hunk 1]
```diff
@@ -261,35 +314,417 @@ struct caloClusterProducerTask {
...
+      if (colbc == colMap.end()) { // single collision per BC
+        colMap[cl.bc_as<aod::BCsWithTimestamps>().globalBC()] = colId;
+      } else { // not unique collision per BC
+        auto coll2 = colls.begin() + colbc->second;
+        if (cl.numContrib() > coll2.numContrib()) {
+          colMap[cl.bc_as<aod::BCsWithTimestamps>().globalBC()] = colId;
+        }
+      }
+      colId++;
+    }
```
Comment (mhemmer-cern) [inline Common/TableProducer/caloClusterProducer.cxx:414]: Why chose the collision with the highest BC?
Why not sliceBy of the collision table joined event selection with o2::aod::evsel::FoundBCId to get a collision for each bc?

File: Common/TableProducer/caloClusterProducer.cxx [hunk 2]
```diff
@@ -261,35 +314,417 @@ struct caloClusterProducerTask {
...
+      cpvNMatchPoints.back().mEnd[i] = cpvMatchPoints[i].size();
+    }
+    // same for tracks
+    std::vector<trackMatch> trackMatchPoints[kCpvCells]; // tracks hit in grid/cell in PHOS
+    // Number of entries in each cell per TrigRecord
+    std::vector<trackTrigRec> trackNMatchPoints;
+    trackNMatchPoints.reserve(outputPHOSClusterTrigRecs.size());
+    trackNMatchPoints.emplace_back();
+
+    curBC = tracks.begin().collision().bc_as<aod::BCsWithTimestamps>().globalBC();
```
Comment (mhemmer-cern) [inline Common/TableProducer/caloClusterProducer.cxx:530]: When using the sliceBy mentioned in my other comment, you could have a loop over collision that are in the current BC (should be one collision) and than use sliceBy to get the tracks associated with that collision and simply loop over these tracks.

Comment (peressounko) [reply Common/TableProducer/caloClusterProducer.cxx:530]: Yes, this is how EMCAL works, but my approach fits PHOS clusterization better.

**Pull request merged**